### PR TITLE
Constrain surface area calculation for Float32 

### PR DESCRIPTION
**Pull request by LenkaNovak**
Closes issue #271 

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/Regridder.jl [hunk 1]
```diff
@@ -409,14 +409,15 @@ function update_masks!(cs::CoupledSimulation)
...
     land_s = cs.surface_masks.land
 
-    cs.surface_masks.ice .= min.(ice_d .+ land_s, FT(1)) .- land_s
-    cs.surface_masks.ocean .= (FT(1) .- cs.surface_masks.ice .- land_s)
+    cs.surface_masks.ice .= min.(ice_d, FT(1) .- land_s)
+    cs.surface_masks.ocean .= max.(FT(1) .- (cs.surface_masks.ice .+ land_s), FT(0)) # max needed to avoid Float32 errors
 
     @assert minimum(cs.surface_masks.ice) >= FT(0)
     @assert minimum(cs.surface_masks.land) >= FT(0)
     @assert minimum(cs.surface_masks.ocean) >= FT(0)
```
Comment (akshaysridhar) [inline src/Regridder.jl:417]: LGTM - maybe include a snippet of the code to show what the problem was initially (generated `-2.98e-8` values for the `cs.surface_masks.ocean` variable while `eps(Float32)` is `1.1920929f-7`, unless this is already tracked elsewhere in an issue). The assertion could be relaxed with `eps(FloatType)` criterion but it looks like we want strict >=0 FT(0) checks here.

Comment (akshaysridhar) [reply src/Regridder.jl:417]: Linked Issue #271

Comment (LenkaNovak) [reply src/Regridder.jl:417]: Sounds good, thank you for the review! I already referred to #271 in the buildkite pipeline file, but it's a great idea to comment in the `Regridder.jl` too.

**Pull request merged**